model-optimizer/extensions/middle/DropoutWithRandomUniformReplacer.py

```diff
@@ -0,0 +1,98 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from typing import Dict
+
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.broadcast import Broadcast
+
+
+class DropoutWithRandomUniformReplacer(MiddleReplacementPattern):
+    """
+    Dropout (for inference mode) with RandomUniform can be replaced with Broadcast of ones in a sub-graph.
+
+    Dropout block:
+    ShapeOf -> RandomUniform -> Mul ---> Add ---> Add -> Floor
+                                /\       /\       /\
+                                |        |        |
+                            Const(1)   Const(0)  Const(1)
+
+    Resulted block:
+    ShapeOf -> Broadcast
+                   /\
+                   |
+               Const(1)
+    """
+    enabled = True
+
+    @staticmethod
+    def pattern():
+        return dict(
+            nodes=[
+                ('shape', dict(op='ShapeOf')),
+                ('shape_data', dict()),
+                ('random_uniform', dict(op='RandomUniform')),
+                ('random_uniform_data', dict()),
+                ('mul', dict(op='Mul')),
+                ('mul_data', dict()),
+                ('mul_const', dict(op='Const')),
+                ('mul_const_data', dict(value=lambda v: v is not None and np.allclose(v, 1.0, atol=1e-6))),
```

Comment:
`atol` should be equal to `0`. Because in this particular case we expect exactly 1.0 value. The same comment for line 60 and 64

Reply:
Fixed.